tests/functionality_tests/obstacle_tracking_test.py

```diff
@@ -182,12 +186,14 @@ def test_min_value(self):
                  math.cos(math.radians((180 - r2d2.sensor_measuring_angle) / 2)) + r2d2.front_sensor_offset))
 
     def test_detection_range(self):
-        wide_robot = Robot(0, 0, math.pi / 2, epsilon=0.2, max_v=0.5, radius=0.2, init_phase=2, width=800,
-                           front_ultrasonic=None, lb_ultrasonic=None, lf_ultrasonic=None, rb_ultrasonic=None,
-                           rf_ultrasonic=None)
-        skinny_robot = Robot(0, 0, math.pi / 2, epsilon=0.2, max_v=0.5, radius=0.2, init_phase=2, width=200,
-                             front_ultrasonic=None, lb_ultrasonic=None, lf_ultrasonic=None, rb_ultrasonic=None,
-                             rf_ultrasonic=None)
+        wide_robot_state = Robot_State(x_pos=0, y_pos=0, heading=math.pi/2, epsilon=0.2, max_velocity=0.5, radius=0.2)
+        wide_robot_state.phase = 2
+        wide_robot_state.width = 800
+        wide_robot = Robot(wide_robot_state)
+        skinny_robot_state = Robot_State(x_pos=0, y_pos=0, heading=math.pi/2, epsilon=0.2, max_velocity=0.5, radius=0.2)
+        skinny_robot_state.phase = 2
+        skinny_robot_state.width = 200
+        skinny_robot = Robot(skinny_robot_state)
```

Comment:
I think the only thing that is needed is changing the width? So maybe smthg like wide_width = 800, skinny_width = 200, robot_state.width = skinny_width or smthg like that to unduplicated

Reply:
a fast review

Reply:
Not sure if this is correct with Python but if we share the robot_state, I ~think~ that they would share the same reference and any changes to the attribute width would affect both the wide_robot and skinny_robot's attributes?

For example, if wide_robot was initialized with robot_state and set to wide_width and then skinny_robot was initialized using that same robot_state, setting the attribute to skinny_width would change robot_state (and thus wide_robot.robot_state is skinny_width)

Reply:
im not sure about this as well

Reply:
hmm i think its objects are passed by reference since they're mutable?
https://www.tutorialspoint.com/how-are-arguments-passed-by-value-or-by-reference-in-python

Reply:
ok update i ran it with using one state and it passed BUT i realized w kwargs it'll prob make more sense readability-wise to have individual states for them